wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Poll/Editor.ts

```diff
@@ -0,0 +1,404 @@
+    this.sortByVotesNoField.checked = true;
+
+    EventHandler.fire(
+      'com.woltlab.wcf.poll.editor',
+      'reset',
+      {
+        pollEditor: this,
+      }
+    );
+  }
+
+  /**
+   * Handles the poll data if the form is submitted.
+   */
+  private submit(event: Event): void {
+    if (this.options.isAjax) {
+      EventHandler.fire(
+        'com.woltlab.wcf.poll.editor',
+        'submit',
+        {
+          event: event,
+          pollEditor: this,
+        }
+      );
+    } else {
+      const form = this.container.closest('form')!;
+
+      const options = this.getOptions();
+      this.getOptions().forEach((option, i) => {
+        const input = document.createElement('input') as HTMLInputElement;
+        input.type = 'hidden';
+        input.name = this.wysiwygId + 'Poll_options[' + i + ']';
+        input.value = option;
+        form.appendChild(input);
+      });
+    }
+  }
+
+  /**
+   * Validates the poll data.
+   */
+  private validate(data: ValidationData): void {
+    if (this.questionField.value.trim() === '') {
+      return;
+    }
+
+    let nonEmptyOptionCount = 0;
+    for (let i = 0, length = this.optionList.children.length; i < length; i++) {
+      const optionInput = this.optionList.children[i].querySelector('input[type=text]') as HTMLInputElement;
+      if (optionInput.value.trim() !== '') {
+        nonEmptyOptionCount++;
+      }
+    }
+
+    if (nonEmptyOptionCount === 0) {
+      data.api.throwError(this.container, Language.get('wcf.global.form.error.empty'));
+      data.valid = false;
+    } else {
+      const maxVotes = ~~this.maxVotesField.value;
+
+      if (maxVotes && maxVotes > nonEmptyOptionCount) {
+        data.api.throwError(this.maxVotesField.parentElement, Language.get('wcf.poll.maxVotes.error.invalid'));
+        data.valid = false;
+      } else {
+        EventHandler.fire(
+          'com.woltlab.wcf.poll.editor',
+          'validate',
+          {
+            data: data,
+            pollEditor: this
+          }
+        );
+      }
+    }
+  }
+
+  /**
+   * Returns the data of the poll.
+   */
+  public getData(): object {
+    let data = {};
+
+    data[this.questionField.id] = this.questionField.value;
+    data[this.wysiwygId + 'Poll_options'] = this.getOptions();
+    data[this.endTimeField.id] = this.endTimeField.value;
+    data[this.maxVotesField.id] = this.maxVotesField.value;
+    data[this.isChangeableYesField.id] = !!this.isChangeableYesField.checked;
+    data[this.isPublicYesField.id] = !!this.isPublicYesField.checked;
+    data[this.resultsRequireVoteYesField.id] = !!this.resultsRequireVoteYesField.checked;
+    data[this.sortByVotesYesField.id] = !!this.sortByVotesYesField.checked;
+
+    return data;
+  }
+
+  /**
+   * Returns the selectable options in the poll.
+   * 
+   * Format: `{optionID}_{option}` with `optionID = 0` if it is a new option.
+   */
+  public getOptions(): string[] {
+    let options = [] as string[];
```

Comment:
```suggestion
    let options: string[] = [];
```
Type declarations are preferred over casts.